services/src/unit-test/java/uk/gov/moj/sdt/services/TargetApplicationSubmissionServiceTest.java

```diff
@@ -145,160 +154,238 @@ public void setUp() {
+        receiveTimeOutParam.setName(TARGET_APP_RESP_TIMEOUT);
+        receiveTimeOutParam.setValue(TWELVE_THOUSAND);
+        when(this.mockCacheable.getValue(IGlobalParameter.class, TARGET_APP_RESP_TIMEOUT)).thenReturn(
                 receiveTimeOutParam);
 
         this.mockConsumerGateway.individualRequest(individualRequest, 1000, 12000);
-        EasyMock.expectLastCall().andAnswer(new IAnswer<Object>() {
-            @Override
-            public Object answer() throws Throwable {
-                ((IndividualRequest) EasyMock.getCurrentArguments()[0])
-                        .setRequestStatus(IIndividualRequest.IndividualRequestStatus.ACCEPTED.getStatus());
-                // required to be null for a void method
-                return null;
-            }
-        });
-
-        this.mockIndividualRequestDao.persist(individualRequest);
-        EasyMock.expectLastCall();
+        doAnswer(invocation -> {
+            IndividualRequest argument = invocation.getArgument(0);
+            argument.setRequestStatus(IIndividualRequest.IndividualRequestStatus.ACCEPTED.getStatus());
+            return null;
+        }).when(mockConsumerGateway).individualRequest(individualRequest, 1000, 12000);
 
         final IBulkSubmission bulkSubmission = individualRequest.getBulkSubmission();
 
-        EasyMock.expect(this.mockIndividualRequestDao.queryAsCount(EasyMock.same(IndividualRequest.class), EasyMock.isA(
-            Supplier.class))).andReturn(0L);
+        when(this.mockIndividualRequestDao.queryAsCount(eq(IndividualRequest.class), isA(
+            Supplier.class))).thenReturn(0L);
 
         mockIndividualRequestDao.persist(bulkSubmission);
-        EasyMock.expectLastCall();
 
-        EasyMock.replay(mockIndividualRequestDao);
-        EasyMock.replay(mockConsumerGateway);
-        EasyMock.replay(mockCacheable);
+        // Setup dummy target response
+        SdtContext.getContext().setRawInXml("response");
+
+        this.targetAppSubmissionService.processRequestToSubmit(sdtRequestRef);
+
+        List<ILoggingEvent> logList = listAppender.list;
+
+        assertNotNull(logList.get(0).getFormattedMessage());
+
+        logger.detachAndStopAllAppenders();
+
+        // Verify the Mock
+        verify(mockIndividualRequestDao,times(2)).persist(individualRequest);
+        verify(mockConsumerGateway, times(2)).individualRequest(individualRequest, 1000, 12000);
+        assertEquals(IBulkSubmission.BulkRequestStatus.COMPLETED
+                .getStatus(), individualRequest.getBulkSubmission().getSubmissionStatus(),"Bulk submission status is incorrect");
+        assertNotNull( individualRequest.getBulkSubmission().getCompletedDate(),
+                       "Bulk submission completed date should be populated");
+        assertNotNull(individualRequest.getBulkSubmission().getUpdatedDate(),
+                      "Bulk submission update date should be populated");
+    }
+
+
```

Comment:
Remove one of these blank lines

Reply:
done

Reply:
There are still double blanks lines before and after the processRequestToSubmitAllSuccessLogSetToError() method.

Reply:
Done